src/main/java/org/mockito/internal/stubbing/defaultanswers/ReturnsEmptyValues.java

```diff
@@ -109,8 +112,20 @@ Object returnValueFor(Class<?> type) {
             return new LinkedHashMap<Object, Object>();
         } else if ("java.util.Optional".equals(type.getName())) {
             return JavaEightUtil.emptyOptional();
+        } else if ("java.util.OptionalDouble".equals(type.getName())) {
```

Comment:
Well it's a behavioural change, so by definition one can write a test that passed prior but not afterwards. For this particular change that'd look something like the following:

1. One has a class `C` with a method `m` returning `OptionalDouble`.
2. One mocks `C`, producing an instance `c`.
3. One fails to explicitly mock the behaviour of `c.m`.
4. One uses `c` in some code that invokes `c.m` and relies on the result being `null`.

I think the "breakage" in this case is acceptable for the following reasons:

1. Since values of type `Optional*` should never be `null`, the failure points out a misuse of `Optional*`, and as such does the user a favour.
2. If one insists on a "nullable `Optional*`", adding a simple mock statement in step 3 above clearly documents that contract, consistent with the [principle of least astonishment](https://en.wikipedia.org/wiki/Principle_of_least_astonishment).
3. More generally, if a test crucially relies on a certain method's return value, then it's good practise to document this precondition explicitly.

I feel the above arguments apply to `Stream*` as much as they do to `Optional*`, because a stream is a (proto) collection in some sense, and nullable collections are also decidedly an anti-pattern. WDYT?